### PR TITLE
[FIX] fieldservice: Keep type when changing template

### PR DESCRIPTION
**Pull request by max3903**

### CODE REVIEW

**Comment by murtuzasaleh** [comment]
LGTM

**Review by murtuzasaleh — APPROVED** [review]

File: fieldservice/models/fsm_order.py [hunk 1]
```diff
@@ -367,6 +367,7 @@ def _onchange_template_id(self):
             self.category_ids = self.template_id.category_ids
             self.scheduled_duration = self.template_id.hours
             self.copy_notes()
-            self.type = self.template_id.type_id
+            if not self.type and self.template_id.type_id:
```
Comment (brian10048) [inline fieldservice/models/fsm_order.py]:
```suggestion
            if self.template_id.type_id:
```

To me the template should set the type unless the template doesn't have a type

Comment (max3903) [reply fieldservice/models/fsm_order.py]: @patrickrwilson What do you think?

Comment (patrickrwilson) [reply fieldservice/models/fsm_order.py]: @max3903 Yes, I agree, if the type field is empty on the fso, and the template has a type, then the type should populate from the template.

**Review by patrickrwilson — APPROVED** [review]
LGTM

**Pull request merged**